app/controllers/application_controller.rb

```diff
@@ -0,0 +1,84 @@
+require 'rubygems'
+require 'bundler/setup'
+require 'sinatra/base'
+
+Bundler.require(:default) 
+
+configure :test do
+ set :database, 'postgresql://localhost/test'
+ DB = Sequel.connect("postgresql://localhost/test")
+end
+
+configure :development do
+  set :database, 'postgresql://localhost/users'
+  DB = Sequel.connect("postgresql://localhost/users")
+end
+
+class ApplicationController < Sinatra::Base
+  enable :sessions
+  enable :method_override
+  disable :show_exceptions   # disabled to hide NoMethodError. Display an error 500 page instead.
+  register Sinatra::Flash
+  set :views, File.expand_path('../../views', __FILE__)
+
+  # create @users variable to give access to view files
+  #-----------------------------------------------------------------
+  def initialize
+    super
+    @users = DB[:users] #=> Sequel::Dataset
+  end
+
+  def user_signed_in?
+    !!session[:current_user_id]
+  end
+
+  def current_user
+    @current_user ||= @users.where(id: session[:current_user_id]).first
+  end
+
+  error 403 do
+    'Access forbidden'
+  end
+
+  # Raises an error if admin goes to a '/users/:id' page that does not exist
+  # Example: '/users/99' will not exist if there is no ID#99 in the database 
+  error 500 do
```

Comment:
While logged in as an admin, I found a NoMethodError when I try to access a page that does not exist because the user does not exist. I used `disable :show_exceptions` to hide the NoMethodError page and instead display an error 500 instead.